lib/engine/game/g_18_esp/game.rb

```diff
@@ -1248,14 +1242,13 @@ def delete_token_mz(minor)
         end
 
         def swap_token(survivor, nonsurvivor)
-          new_token = survivor.tokens.last
+          new_token = survivor.next_token
```

Comment:
Is it possible that `survivor` will not have any remaining tokens?

Reply:
No, as the method calling this ( start_merge(corporation, minor, keep_token)), is called by the acquire step, where the ability to swap is checked in can_swap? there, which checks for next_token of survivor.